ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +219,29 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)
+          dir? (jar-directory? conn)]
+      {:content        (if-not dir? (.getInputStream conn))
```

Comment:
Maybe this could more concisely be written as:

```clojure
(let [conn (.openConnection url)]
  (if (jar-directory? conn)
    {:directory true}
    {:directory? false
     :content (.getInputStream conn)
     :content-length (jar-content-length conn)
     :last-modified  (jar-last-modified conn)}))
```

However, now I see it written out like that, I wonder if we need a `:directory?` key. If this is for delivering static resources, then if the resource is a directory, we should always ignore it. So perhaps:

```clojure
(let [conn (.openConnection url)]
  (if-not (jar-directory? conn)
    {:content (.getInputStream conn)
     :content-length (jar-content-length conn)
     :last-modified  (jar-last-modified conn)}))
```

What do you think?